### DISCUSSION TITLE
Monthly ESMValTool community meeting July 2022

### DISCUSSION THREAD

**Discussion by bouweandela** [ESMValGroup/Community#33] [Meetings]
I realize that the monthly meetings have not quite been monthly recently, but from September onwards @bettina-gier will take over running these meetings, so we can all look forward to a more regular schedule and well-organized meetings from then on.

The next monthly ESMValTool meeting will be in July. The result of the poll is that the meeting will take place on

Monday, July 11 at 15:00 CEST (14:00 BST).

The duration will be one hour. [Join the zoom meeting](https://us02web.zoom.us/j/86356385817).

Calendar invitations to the meeting are sent to the mailing list, join it [here](https://docs.esmvaltool.org/en/latest/introduction.html#user-mailing-list).

https://github.com/orgs/ESMValGroup/teams/esmvaltool-developmentteam

Format and agenda

- Anyone with an interest in ESMValTool is welcome to join and ask questions
- We start the meeting by doing a round where everyone gets one minute to introduce themselves and tell what they will be working on in the next month or what they would like help with.
- Using the input from the introduction round, we can decide to talk about specific topics with the entire group or make smaller breakout groups to continue the discussion there.
- If you already know that you would like to bring up a particular topic in the meeting, you can mention the [GitHub issue](https://github.com/ESMValGroup/ESMValTool/issues) on the topic in a comment below. That will make it easy to find for other people in the meeting and we can already start discussing there before the meeting. Just create a new GitHub issue about your topic if there is no open issue yet.
- GitHub issues are our preferred way to communicate outside the meeting and keep track of work, so any feature requests/action items/etc discussed in the meeting should ideally result in a GitHub issue.

**Comment by AnushaNTNU** [comment]
CMIP6 model comparisons using ESMValTool

**Reply by bouweandela** [reply]
@AnushaNTNU We missed you at the meeting. What did you want to discuss about CMIP6 model comparisons using ESMValTool? Perhaps about updating recipes that currently still use CMIP5 data?

**Reply by AnushaNTNU** [reply]
Hi, I just started using ESMValTool version , installed at the  IPCC node of NIRD supercomputer here in Norway. I succesfully ran a sample recipe today. i would like to know the location of the above said recipes which are using CMIP5 data. And cant the user update it while using?

**Reply by bouweandela** [reply]
>  I succesfully ran a sample recipe today.

:partying_face: 

> i would like to know the location of the above said recipes which are using CMIP5 data. 

Have a look at the `esmvaltool recipes` command and the recipes documentation. You can read all about it here: https://docs.esmvaltool.org/en/latest/quickstart/running.html#available-diagnostics-and-metrics

> And cant the user update it while using?

This usually works fine for the preprocessor part of the recipe (though you may need to add [fixes](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/develop/fixing_data.html) if you're trying to use a dataset that has some issues and has not been used before by anyone), but sometimes the diagnostic scripts depend on really specific things. I would recommend just trying it and seeing if it works with the recipe that you're interested in. If it doesn't work, you can do an [installation from source](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#install-from-source) so you can modify the code where needed.

**Comment by glpotter** [comment]
Adding 3D variables to cmorization

**Comment by bouweandela** [comment]
# Meeting notes

Thanks for joining everyone!

## Here are some personal updates:
@remi-kazeroni Has been working on moving the esmvaltool.org website to github. Currently the code is here: https://github.com/ESMValGroup/ESMValGroup.github.io.
@malininae Joined for the first time, working on attribution of extreme events: struggling with using preprocessor on big data, especially long time series
@glpotter Retired, but interested in CMORizing data for CREATE-IP project. This looks like a very useful project for the ESMValTool community and may also help with some of the issues we have with (not) distributing CMORized data. See also ESMValGroup/ESMValCore#1652.
@ehogan Porting AutoAssess software to ESMValTool
@lukruh Will be working with ICON model data and is doing a PhD on droughts
@mo-gregmunday Opened ESMValGroup/ESMValTool#2702 with a recipe on climate patterns, looking for scientific reviewer. See also the advice in our documentation on [finding reviewers](https://docs.esmvaltool.org/en/latest/community/review.html#how-do-i-request-a-review-of-my-pull-request).
@mo-tgeddes Industrial Placement working with @ehogan

Please feel free to change/update the above if I missed something.

## We discussed these topics:
- Tutorial webpage https://github.com/ESMValGroup/Community/discussions/33#discussioncomment-3121404
- Preprocessor https://github.com/ESMValGroup/Community/discussions/33#discussioncomment-3121419
- CMORization of MERRA2 https://github.com/ESMValGroup/Community/discussions/33#discussioncomment-3121558
- fx variables https://github.com/ESMValGroup/Community/discussions/33#discussioncomment-3121606
- CaPA reanalysis CMORization https://github.com/ESMValGroup/Community/discussions/33#discussioncomment-3121663

**Comment by bouweandela** [comment]
Tutorial webpage, how to host both the tutorial webpage and the main esmvaltool.org webpage on github?

Suggestions were to look at:
- https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain 
- is-enes webpage setup on github
- This stackoverflow answer may also be helpful: https://stackoverflow.com/q/15563685

**Reply by bouweandela** [reply]
Maybe the main website could be hosted at [esmvaltool.org](https://esmvaltool.org) and the tutorial(s) at [tutorials.esmvaltool.org](https://tutorials.esmvaltool.org)?

**Reply by remi-kazeroni** [reply]
Issue available at: https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/150

**Comment by bouweandela** [comment]
Preprocessor unable to handle large datasets: @malininae will create an issue and share an example recipe

**Reply by bouweandela** [reply]
Suggestion from @zklaus: using one of the new regridding schemes might help (maybe): https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/preprocessor.html#generic-regridding-schemes

**Reply by malininae** [reply]
> Preprocessor unable to handle large datasets: @malininae will create an issue and share an example recipe

Wanted to post an update, my organisation moved to a different HPC system as of  the 1st of July, and it seems that the issues I had before are not anymore reproducible. So for now, it seems that the preprocessor even with a bunch of operations, including regridding and anomalies on a daily data works.  Sorry for raising an old issue! If I will it ever encounter again, I will post it!

**Comment by bouweandela** [comment]
CMORization of MERRA2

@glpotter Has no previous experience with iris and has until October. Would like some help. The idea would be to point users of the CREATE-IP project to the ESMValTool CMORizer.

@glpotter Will create an issue and @valeriupredoi offered to help out with technical points.

**Reply by bouweandela** [reply]
Issue created here: https://github.com/ESMValGroup/ESMValTool/issues/2718

**Comment by bouweandela** [comment]
Finding and adding fx variables `sftlf`. 

- Recipe documentation: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/recipe/preprocessor.html#fx-variables-as-cell-measures-or-ancillary-variables 
- Searching ESGF: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.esgf.html#esmvalcore.esgf.find_files (note the typo in the documentation, the function is called `find_files`, not `search`).
- Example of a script that searches for datasets that have two variables available: https://github.com/ESMValGroup/ESMValCore/issues/1484. A variation of this could be used to find models that have both the required variable and sftlf available.
- Pull request that improves support for specifying fx variables in the recipe: https://github.com/ESMValGroup/ESMValCore/pull/1609

**Comment by bouweandela** [comment]
CaPA precipitation reanalysis CMORization

Special grid. @malininae Will create an issue with a link to the data and ask for help there.

**Reply by bouweandela** [reply]
As long as it is possible to convert the data to something that follows the [cf conventions](https://cfconventions.org/cf-conventions/cf-conventions.html#_two_dimensional_latitude_longitude_coordinate_variables), it seems likely that it can be CMORized.

**Comment by AnushaNTNU** [comment]
Hi

I haven't started using ESMValTool actually. But I would really like to use it sooner, I got access to the IPCC node of NIRD supercomputer here in Norway recently where the ESMValTool software is installed. I would like to attend any workshop if available to get started with ESMValTool.


Thanks,
Anusha




________________________________

**Reply by bouweandela** [reply]
@ESMValGroup/userengagementteam Do we have any tutorials coming up?

**Reply by LisaBock** [reply]
Hi @AnushaNTNU 

I don't know of any tutorials in the near future. Next April a short course is planned at the EGU.

You should also be able to work through the [ESMValTool Tutorial](https://esmvalgroup.github.io/ESMValTool_Tutorial/) on your own. It is designed therefor.

**Reply by AnushaNTNU** [reply]
Thanks a lot. I will follow that
